types/providers.d.ts

```diff
@@ -29,13 +29,13 @@ export interface OAuthConfig<P extends Record<string, unknown> = Profile>
   scope: string
   params: { grant_type: string }
   accessTokenUrl: string
-  requestTokenUrl: string
+  requestTokenUrl?: string
   authorizationUrl: string
-  profileUrl: string
-  profile(profile: P, tokens: TokenSet): Awaitable<User & { id: string }>
+  profileUrl?: string
+  profile?(profile: P, tokens: TokenSet): Awaitable<User & { id: string }>
   protection?: ProtectionType | ProtectionType[]
   clientId: string
-  clientSecret:
+  clientSecret?:
```

Comment:
I cannot think of a single OAuth provider where this could be optional. Do you have an example? If it says in the docs that this is optional, I think it should be fixed there!

Reply:
Yes, you're right, it didn't make much sense to me either 😅
But then I just trusted the documentation...

I'm going to reverse this and correct the docs, if that's ok.